core/Console.php

```diff
@@ -209,6 +209,35 @@ private function getDefaultPiwikCommands()
         $commands = array(
             'Piwik\CliMulti\RequestCommand'
         );
+
+        $commandsFromPluginsMarkedInConfig = $this->getCommandsFromPluginsMarkedInConfig();
+        if(!empty($commandsFromPluginsMarkedInConfig)) {
+            $commands = array_merge($commands, $commandsFromPluginsMarkedInConfig);
+        }
+
+        return $commands;
+    }
+
+    private function getCommandsFromPluginsMarkedInConfig()
+    {
+        $general = Config::getInstance()->getFromCommonConfig('General');
+
+        if (empty($general)) {
+            return null;
```

Comment:
I haven't tested it but looks good I reckon. If you returned an empty array here and below you wouldn't need the check above for `!empty($commandsFromPluginsMarkedInConfig)` and code would be bit easier to read / cleaner. Ideally we always return the same type eg always `array`.